buildSrc/src/main/java/org/elasticsearch/gradle/VersionCollection.java

```diff
@@ -186,7 +183,7 @@ public void forPreviousUnreleased(Consumer<UnreleasedVersionInfo> consumer) {
 
     private String getGradleProjectNameFor(Version version) {
```

Comment:
While @jasontedor and I were walking through this together we found it really hard to reason about this logic. I later ready the top-level javadoc and that cleared a few things up but I think in general this method in particular could use some comments to guide the reader along. Can we add some comments to the logic in this method to better explain what is happening?

Reply:
I added a few comments and a pointer to the class javadoc